accio-sqlrewrite/src/test/java/io/accio/sqlrewrite/TestExpressionRelationshipRewriter.java

```diff
@@ -0,0 +1,106 @@
+    }
+
+    @DataProvider
+    public Object[][] rewriteTests()
+    {
+        return new Object[][] {
+                {"customer.custkey", "\"Customer\".\"custkey\"", List.of(ordersCustomer)},
+                {"customer.nation.name", "\"Nation\".\"name\"", List.of(ordersCustomer, customerNation)},
+                {"customer.nation.nationkey + 1", "(\"Nation\".\"nationkey\" + 1)", List.of(ordersCustomer, customerNation)},
+                {"concat('#', customer.nation.name)", "concat('#', \"Nation\".\"name\")", List.of(ordersCustomer, customerNation)},
+                {"concat(customer.name, '#', customer.nation.name)", "concat(\"Customer\".\"name\", '#', \"Nation\".\"name\")",
+                        List.of(ordersCustomer, customerNation, ordersCustomer)},
+        };
+    }
+
+    @Test(dataProvider = "rewriteTests")
+    public void testRewrite(String actual, String expected, List<Relationship> relationships)
+    {
+        Expression expression = parseExpression(actual);
+        List<ExpressionRelationshipInfo> expressionRelationshipInfos = ExpressionRelationshipAnalyzer.getRelationships(expression, mdl, orders);
+        assertThat(expressionRelationshipInfos.stream().map(ExpressionRelationshipInfo::getRelationships).flatMap(List::stream).collect(toImmutableList()))
+                .containsExactlyInAnyOrderElementsOf(relationships);
+        assertThat(RelationshipRewriter.rewrite(expressionRelationshipInfos, expression).toString()).isEqualTo(expected);
+    }
+
+    @Test
+    public void testToMany()
+    {
+        assertThatThrownBy(() -> ExpressionRelationshipAnalyzer.getRelationships(parseExpression("customer.custkey"), mdl, nation))
+                .hasMessage("expr in model only accept to-one relation");
+        assertThatThrownBy(() -> ExpressionRelationshipAnalyzer.getRelationships(parseExpression("customer.nation.customer.custkey"), mdl, orders))
+                .hasMessage("expr in model only accept to-one relation");
+    }
+
+    @Test
+    public void testNoRelationshipFound()
+    {
+        // won't collect relationship if direct access relationship column
+        assertThat(ExpressionRelationshipAnalyzer.getRelationships(parseExpression("customer"), mdl, nation)).isEmpty();
+        assertThat(ExpressionRelationshipAnalyzer.getRelationships(parseExpression("customer.nation"), mdl, orders)).isEmpty();
+        // won't collect relationship if column not found in model
+        assertThat(ExpressionRelationshipAnalyzer.getRelationships(parseExpression("foo"), mdl, orders)).isEmpty();
+        // won't collect relationship since "Orders" is not a column in orders model
+        assertThat(ExpressionRelationshipAnalyzer.getRelationships(parseExpression("Orders.customer.custkey"), mdl, orders)).isEmpty();
+    }
+
+    @Test
+    public void testCycle()
+    {
+        assertThatThrownBy(() -> ExpressionRelationshipAnalyzer.getRelationships(parseExpression("region.nation"), mdl, nation))
```

Comment:
Same as above. Let's add a TODO comment for it.

Reply:
This is another scenario, IMO, we shouldn't let user do this. `a.b.a.col_1` in expression.